bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/proxy/AsyncProxyServlet.java

```diff
@@ -51,6 +52,21 @@ protected HttpClient newHttpClient() {
         return new HttpClient(new SslContextFactory());
     }
 
+    @Override
+    protected void sendProxyRequest(HttpServletRequest clientRequest, HttpServletResponse proxyResponse,
+            Request proxyRequest) {
+        if (service.proxyingVideoWidget(clientRequest)) {
+            // We disable the timeout for video
+            proxyRequest.timeout(0, TimeUnit.MILLISECONDS);
+
+            // We request the browser to not cache the video
```

Comment:
The proxy request has not been sent yet.
Are you sure, that it is intended to change the proxy response header already?

Shouldn't the `cache-control` values set to the respective values for all request?
Makes it sense to cache any result?

Reply:
I don't know if the implementation allows intercepting the answer to update its headers. My current implementation allows presetting headers; the headers from the source servers are apparently concatenated with them.

Regarding your question if we should avoid browser caching for images too, that is a good question. In fact all UIs running in browser use a workaround to avoid browser caching, they add a timestamp parameter to the proxied URL. So that could make sense to apply this change for all content (image and video).